Use logs in `Adt_rel`

Halbaroth opened the pull request:
This PR is rebased on #1204 and #1206

Halbaroth committed:
Use logs in `Adt_rel`

bclement-ocp commented on src/lib/reasoners/adt_rel.ml:
```diff
@@ -319,45 +319,9 @@ let empty uf = {
-        "%a" X.print r
-
+  let assume l =
+    Logs.debug ~src:Sources.adt_rel
```
Please don't use `Logs.debug ~src:...`. Instead, define `module Log = (val Logs.src_log Sources.adt_rel)` at the top of the file, and use `Log.debug` directly. See the [example](https://erratique.ch/software/logs/doc/Logs/index.html#setupreporter) from `Logs`'s documentation.

bclement-ocp commented on src/lib/reasoners/adt_rel.ml:
```diff
@@ -319,45 +319,9 @@ let empty uf = {
-  let pp_domains loc domains =
-    if Options.get_debug_adt () then begin
-      print_dbg ~flushed:false ~module_name:"Adt_rel"
-        "@[<v 2>--ADT env %s ---------------------------------@ " loc;
```
Looks like this PR removes the the `---` separators that can make the logs more readable; is that intentional?

Halbaroth replied:
It is intentional because `Logs` uses correctly indentations. So we will get something like that:
```
[DEBUG][Adt_rel] environment before assume:
                 the environment here 
                 .... 
[DEBUG][Adt_rel] ... 
``` 
The indentation makes clear in which messages we are.

Halbaroth committed:
review changes

bclement-ocp reviewed (approved):
LGTM. Please merge #1206 first and rebase before merging.

Halbaroth commented:
Actually I already rebased it and I expected that github will skip empty commit. I rebased it again to be sure but now the CI is dead :D (it is not our fault).
